lib/buffer.js

```diff
@@ -246,18 +247,30 @@ function fromArrayLike(obj) {
   const length = obj.length;
   const b = allocate(length);
   for (var i = 0; i < length; i++)
-    b[i] = obj[i] & 255;
+    b[i] = obj[i];
```

Comment:
Just another unnecessary indirection that slightly changes performance (we inherit from `Uint8Array`, which already performs this exact same normalization).